### PR TITLE
[Debugger] Fix expression evaluation support for lang library sources

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR 
- fixes https://github.com/ballerina-platform/ballerina-lang/issues/34282

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/ModuleLevelDefinitionFinder.java [hunk 1]
```diff
@@ -98,12 +99,25 @@ protected void visitSyntaxNode(Node node) {
...
+        // Ignores external function definitions.
+        if (node instanceof FunctionDefinitionNode && ((FunctionDefinitionNode) node).functionBody().kind() ==
+                SyntaxKind.EXTERNAL_FUNCTION_BODY) {
+            return;
+        }
+
+        // Ignores type definitions with @builtinSubtype annotations (specific to Ballerina library sources).
+        if (node instanceof TypeDefinitionNode && ((TypeDefinitionNode) node).metadata().isPresent() &&
+                ((TypeDefinitionNode) node).metadata().get().annotations().stream().anyMatch(annotationNode ->
+                        annotationNode.annotReference().toSourceCode().trim().equalsIgnoreCase("builtinSubtype"))) {
```
Comment (praveennadarajah) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/ModuleLevelDefinitionFinder.java:117]: Shall we move the hardcoded `builtinSubtype` string to a constant?

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/ModuleLevelDefinitionFinder.java [hunk 2]
```diff
@@ -49,6 +50,8 @@
     private final Set<SyntaxKind> filters = new HashSet<>();
     private final List<NonTerminalNode> result = new ArrayList<>();
 
+    private static final String ANNOTATION_BUITLINSUBTYPE = "builtinSubtype";
```
Comment (praveennadarajah) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/ModuleLevelDefinitionFinder.java:53]:
```suggestion
    private static final String ANNOTATION_BUILTINSUBTYPE = "builtinSubtype";
```

Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/ModuleLevelDefinitionFinder.java]: FIxed

**Review by praveennadarajah — DISMISSED** [review]

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/ModuleLevelDefinitionFinder.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/ModuleLevelDefinitionFinder.java]: Done

**Review by praveennadarajah — APPROVED** [review]

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**